File: 프로그래머스/1/131128. 숫자 짝꿍/숫자 짝꿍.java
import java.io.*;
import java.util.*;
class Solution {
    public String solution(String X, String Y) {
        int arr[][] = new int[10][2];
        for(int i=0; i<X.length(); i++){
            arr[Integer.parseInt(X.substring(i,i+1))][0] ++;
        }
        for(int i=0; i<Y.length(); i++){
            arr[Integer.parseInt(Y.substring(i,i+1))][1] ++;
        }
        StringBuilder sb = new StringBuilder();
        for(int i=9; i>=0; i--){
            int nummin = Math.min(arr[i][0], arr[i][1]);
            for(int j=0; j<nummin; j++){
                sb.append(i);
            }
        }
        String answer = sb.toString();
        if(answer.isEmpty()){
            answer = "-1";
        }
        if(answer.startsWith("0")){
            answer = "0";
        }
        return answer;
    }
}

File: 프로그래머스/1/131128. 숫자 짝꿍/숫자 짝꿍.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string X, string Y) {
    string answer = "";
    int arr1[10] = {};
    int arr2[10] = {};
    for(int i=0; i<X.length(); i++){
        int tmp = X[i] - '0';
        arr1[tmp]++;
    }
    for(int i=0; i<Y.length(); i++){
        int tmp = Y[i] - '0';
        arr2[tmp]++;
    }
    for(int i=9; i>=0; i--){
        int tmp = min(arr1[i], arr2[i]);
        if(i == 0 && tmp > 0 && answer.empty()){
            answer += to_string(0);
        }
        else{
            for(int j=0; j<tmp; j++){
                answer +=  to_string(i);
            }  
        }
    }
    if(answer.empty()){
        answer = "-1";
    }
    return answer;
}